src/array.cr

```diff
@@ -234,7 +234,7 @@ class Array(T)
   # [2] <=> [4, 2, 3] # => -1
   # [1, 2] <=> [1, 2] # => 0
   # ```
-  def <=>(other : Array)
+  def <=>(other : Array) : Int32
     min_size = Math.min(size, other.size)
     0.upto(min_size - 1) do |i|
       n = @buffer[i] <=> other.to_unsafe[i]
```

Comment:
There are cases (like for `Number`) where `<=>` is nilable, so restricting to `Int32` may be incorrect. Especially for different variants of `Float` where comparison against `NaN` gives nil back.

Reply:
We could probably still add `Int32?` as return type restriction.

Reply:
That would be misleading if just Int32 is returned.